File: C Programming/ArrayStack/ArrayStackDS.c
#include <stdio.h>
#include <stdlib.h>

int *mainStack = NULL;
int top = -1;
int size = 0;

void push(int data) {
    printf("\nSUCCESS: pushing(%d) into array stack.", data);
    if (mainStack == NULL) {
        mainStack = (int*) malloc(sizeof(int));
        top = 0;
        mainStack[top] = data;
        size++;
    } else {
        if (top == 0) {
            int *tempStack = (int*) malloc((size + 1) * sizeof(int));
            tempStack[top] = mainStack[top];
            top++;
            tempStack[top] = data;
            mainStack = tempStack;
            size++;
        } else {
            int *tempStack = (int*) malloc((size + 1) * sizeof(int));
            for (int i = 0; i < size; i++) {
                tempStack[i] = mainStack[i];
            }
            top++;
            tempStack[top] = data;
            mainStack = tempStack;
            size++;
        }
    }
}

void peek() {
    if (mainStack == NULL) {
        printf("\nERROR: cannot peek from empty array stack.");
    } else {
        printf("\nmainStack.peek() --> %d\n", mainStack[top]);
    }
}

void pop() {
    if (mainStack == NULL) {
        printf("\nERROR: cannot pop() from empty array stack.");
    } else {
        printf("\nSUCCESS: popping(%d) from array stack.", mainStack[top]);
        if (top == 0) {
            mainStack = NULL;
            top--;
            size--;
        } else {
            int *tempStack = (int*) malloc((size - 1) * sizeof(int));
            size--;
            for (int i = 0; i < size; i++) {
                tempStack[i] = mainStack[i];
            }
            top--;
            mainStack = tempStack;
        }
    }
}

void printArrayStack() {
    if (top == -1) {
        printf("\nTOP\n----\n");
        printf("----\n");
    } else {
        printf("\nTOP\n----\n");
        for (int i = size - 1; i >= 0; i--) {
            printf("%d\n", mainStack[i]);
        }
        printf("----\n");
    }
}

int main(int argc, char* argv[]) {
    printf("ArrayStack.c\n");
    printf("------------\n");
    printArrayStack();
    push(3);
    printArrayStack();
    push(22);
    printArrayStack();
    push(7);
    printArrayStack();
    push(81);
    printArrayStack();
    peek();
    pop();
    printArrayStack();
    pop();
    printArrayStack();
    pop();
    printArrayStack();
    pop();
    printArrayStack();
    return EXIT_SUCCESS;
}

File: Java Programming/ArrayStack/ArrayStackDS.java
public class ArrayStackDS {

    int[] mainStack = null;
    int top = -1;

    public ArrayStackDS() {
        System.out.println("Successful initialization of array stack object.");
    }

    public void push(int data) {
        System.out.println("\nSUCCESS: pushing(" + data + ") into array stack.");
        if (mainStack == null) {
            mainStack = new int[1];
            top = 0;
            mainStack[top] = data;
        } else {
            if (top == 0) {
                int[] tempStack = new int[mainStack.length + 1];
                tempStack[top] = mainStack[top];
                top++;
                tempStack[top] = data;
                mainStack = tempStack;
            } else {
                int[] tempStack = new int[mainStack.length + 1];
                for (int i = 0; i < mainStack.length; i++) {
                    tempStack[i] = mainStack[i];
                }
                top++;
                tempStack[top] = data;
                mainStack = tempStack;
            }
        }
    }

    public void peek() {
        if (mainStack == null) {
            System.out.println("\nERROR: cannot peek from empty array stack.");
        } else {
            System.out.println("\nmainStack.peek() --> " + mainStack[top]);
        }
    }

    public void pop() {
        if (mainStack == null) {
            System.out.println("\nERROR: cannot pop() from empty array stack.");
        } else {
            System.out.println("\nSUCCESS: popping(" + mainStack[top] + ") from array stack.");
            if (top == 0) {
                mainStack = null;
                top--;
            } else {
                int[] tempStack = new int[mainStack.length - 1];
                for (int i = 0; i < tempStack.length; i++) {
                    tempStack[i] = mainStack[i];
                }
                top--;
                mainStack = tempStack;
            }
        }
    }

    public void printArrayStack() {
        if (top == -1) {
            System.out.println("TOP\n----");
            System.out.println("----");
        } else {
            System.out.println("TOP\n----");
            for (int i = mainStack.length - 1; i >= 0; i--) {
                System.out.println(mainStack[i]);
            }
            System.out.println("----");
        }
    }
}
